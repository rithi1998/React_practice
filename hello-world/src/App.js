import logo from './logo.svg';
import './App.css';
import Greet from './components'

function App() {
  return (
    <div className="App">
      <Greet/>
    </div>
  );
}

export {App};
