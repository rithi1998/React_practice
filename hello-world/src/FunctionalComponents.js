import React from 'react'

// function Greet(){
//     return <h1> Hello Rithika</h1>
// }

const Greet=() => <h1>Hi Rithika</h1>

export default Greet