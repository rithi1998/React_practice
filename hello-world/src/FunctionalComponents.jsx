import React from 'react';
import PropTypes from 'prop-types';
// function Greet(){
//     return <h1> Hello Rithika</h1>
// }

const Greet = () => <h1>Hi Rithika</h1>;

export default Greet;


export function Welcome(props) {
    const name = 'Rajkumar';
    return (
        <>
            <h1> Hello Raj</h1>

            <h2>
                Hello {name} {props.greet}
            </h2>
        </>
    );
}

Welcome.propTypes={
    greet:PropTypes.string
};