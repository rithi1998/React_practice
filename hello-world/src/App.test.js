import { render, screen } from '@testing-library/react';
import App from './App';
import FirstBind from './Bind';
import { Welcome } from './FunctionalComponents';
import renderer from 'react-test-renderer';
// test('renders learn react link', () => {
//   render(<FirstBind />);
//   const linkElement = screen.getByTestId("Rithi");
//   expect(linkElement).toBeInTheDocument()
// })

// it('Functional component',()=>{
//   render(<Welcome/>);
//   const FunctionElement=screen.getByText("Hello Raj");
//   expect(FunctionElement).toBeInTheDocument()
// })

test('renders the components',()=>{
  const tree=renderer.create(
    <div>
      <FirstBind/>
      <Welcome/>
    </div>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});