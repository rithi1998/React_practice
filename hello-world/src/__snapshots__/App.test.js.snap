// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the components 1`] = `
<div>
  <button
    data-testid="Rithi"
    onClick={[Function]}
  >
    Click here
  </button>
  <h1>
     Hello Raj
  </h1>
  <h2>
    Hello 
    Rajkumar
     
  </h2>
</div>
`;
